listeners/assistant/message.py

```diff
@@ -34,47 +36,95 @@ def message(
-        for event in returned_message:
-            if event.type == "response.output_text.delta":
-                streamer.append(markdown_text=f"{event.delta}")
-            else:
-                continue
+        # This first example shows a generated text response for the provided prompt
+        if message["text"] != "Wonder a few deep thoughts.":
+            set_status(
+                status="thinking...",
+                loading_messages=[
+                    "Teaching the hamsters to type faster…",
+                    "Untangling the internet cables…",
+                    "Consulting the office goldfish…",
+                    "Polishing up the response just for you…",
+                    "Convincing the AI to stop overthinking…",
+                ],
+            )
+
+            # Loop over OpenAI response stream
+            # https://platform.openai.com/docs/api-reference/responses/create
+            for event in call_llm(message["text"]):
+                if event.type == "response.output_text.delta":
+                    streamer.append(markdown_text=f"{event.delta}")
+                else:
+                    continue
+
+            feedback_block = create_feedback_block()
+            streamer.stop(
+                blocks=feedback_block,
+            )
+
+        # The second example shows detailed thinking steps similar to tool calls
+        else:
+            streamer.append(
+                chunks=[
+                    MarkdownTextChunk(
+                        text="Hello.\nI have received the task. ",
+                    ),
+                    MarkdownTextChunk(
+                        text="This task appears manageable.\nThat is good.",
+                    ),
+                    TaskUpdateChunk(
+                        id="001",
+                        title="Understanding the task...",
+                        status="in_progress",
+                        details="- Identifying the goal\n- Identifying constraints",
+                    ),
+                    TaskUpdateChunk(
+                        id="002",
+                        title="Performing acrobatics...",
+                        status="pending",
+                    ),
+                ],
+            )
+            time.sleep(4)
+
+            streamer.append(
+                chunks=[
+                    TaskUpdateChunk(
+                        id="001",
+                        title="Understanding the task...",
+                        status="complete",
+                        details="\n- Pretending this was obvious",
+                        output="We'll continue to ramble now",
+                    ),
+                    TaskUpdateChunk(
+                        id="002",
+                        title="Performing acrobatics...",
+                        status="in_progress",
+                    ),
+                ],
+            )
+            time.sleep(4)
```

Comment:
Nice! I've been experimenting with something similar on my own,  I like this idea. Featuring what is the agent doing is great like (tools calls or thinking steps)